src/app/utils/helpers.tsx

```diff
@@ -0,0 +1,15 @@
+import {
+  ResourceType,
+  ResourceMap,
+} from "../patient-summary/types/resources.types";
+
+export const extractResourceInfo = <T extends ResourceType>(
+  resourceType: T,
+  data: any
+): T extends keyof ResourceMap ? ResourceMap[T] : undefined => {
+  const resource = data.entry?.find(
```

Comment:
is this  meant to return the first entry of the specified resource type? is it possible for the bundle to contain multiple entries of a resource type. In this case what would be the conditions for selecting a specific one? Perhaps split the logic. First get all entries of the said resource type then filter based on a condition.

Reply:
Hello, 
Based on the FHIR specification, a bundle cannot contain multiple entries with the same resource type under the same resource key. 
This means that the first entry returned by the find method is indeed the only entry with the selected resource type.